apps/native-component-list/src/screens/UI/ColorPickerScreen.tsx

```diff
@@ -0,0 +1,41 @@
+import { ColorPicker } from '@expo/ui/components/ColorPicker';
+import * as React from 'react';
+import { Dimensions, ScrollView, Text } from 'react-native';
+
+import { Page, Section } from '../../components/Page';
+import { Switch } from '@expo/ui/components/Switch';
+
+export default function ColorPickerScreen() {
+  const [color, setColor] = React.useState<string | null>('blue');
+  const [supportsOpacity, setSupportsOpacity] = React.useState<boolean>(false);
+  return (
+    <ScrollView>
+      <Page>
+        <Section title="Color Picker">
+          <Text>Color: {color}</Text>
+          <Switch
+            label="Supports opacity"
+            checked={supportsOpacity}
+            onCheckedChanged={({ nativeEvent: { checked } }) => setSupportsOpacity(checked)}
+          />
+          <ColorPicker
+            label="Select a color"
+            selection={color}
+            supportsOpacity={supportsOpacity}
+            onValueChanged={({ nativeEvent: { value } }) => {
```

Comment:
We should unpack the `nativeEvent` in the component. This was an oversight, as it wasn't done in other UI components. We did it inside of `expo-maps` - see https://github.com/expo/expo/blob/757c2d4dcff552244b5cfb34c54fe18887ff6c98/packages/expo-maps/src/google/GoogleMapsView.tsx#L17-L24

To give you more context, we don't want to use our API's `nativeEvent` key. It's a React Native API, and we don't have any control over it. If they change it, users need to migrate. By unpacking it in our code, we can migrate if needed. Also, in the long run, we would like to migrate the underlying event emitter to our C++ implementation. Currently, we're using one from RN for views.

Reply:
I agree, thanks. Updated!